### PR TITLE
Use Redix

### PR DESCRIPTION
**Pull request by mitchellhenke**
So there are some issues yet to work out, but this is my try at replacing redo with redix.

The biggest issue is behavior on startup when redis can't be reached.  Right now, if you start an app using these changes, the app will crash (where the previous redo version would not).  It's related to the call on init to connect, and I'm not sure what structure would better handle it.

Redix always returns `{:ok, pid}` [here](https://github.com/mitchellhenke/phoenix_pubsub_redis/blob/redix/lib/phoenix_pubsub_redis/redis_server.ex#L108) (even if it's not), and this times out because the process returned goes down immediately, causing it to fail [here](https://github.com/mitchellhenke/phoenix_pubsub_redis/blob/redix/lib/phoenix_pubsub_redis/redis_server.ex#L102), and so `Phoenix.PubSub.RedisServer` crashes the entire tree on start since it never finishes starting.

I hope I'm headed somewhat in the right direction, and would be happy to keep trying if you have time to help me through the aforementioned issue, as well as any others you see.

### CODE REVIEW

**Comment by mitchellhenke** [comment]
@jeregrine and I got past it by using sync_connect and using the existing retry timers

**Comment by jeregrine** [comment]
@chrismccord ^^ Ready for review.

File: lib/phoenix_pubsub_redis/redis_server.ex [hunk 1]
```diff
@@ -24,46 +25,43 @@ defmodule Phoenix.PubSub.RedisServer do
...
-      case GenServer.call(worker_pid, :conn) do
-        {:ok, conn_pid} ->
-          case :redo.cmd(conn_pid, ["PUBLISH", namespace, bin_msg]) do
-            {:error, reason} -> {:error, reason}
-            _ -> :ok
-          end
-
+      case Redix.command(worker_pid, ["PUBLISH", namespace, bin_msg]) do
         {:error, reason} -> {:error, reason}
+        _ -> :ok
```
Comment (chrismccord) [inline lib/phoenix_pubsub_redis/redis_server.ex]: Let's match on the success case(s) explicitly

File: lib/phoenix_pubsub_redis/redis_server.ex [hunk 2]
```diff
@@ -134,11 +104,17 @@ defmodule Phoenix.PubSub.RedisServer do
...
-      _error          -> establish_failed(state)
-    end
+  defp establish_success(%{redix_pid: redix_pid} = state) do
+    :ok = Redix.PubSub.subscribe(redix_pid, state.namespace, self())
+    %{state | status: :connected}
   end
 
+  defp establish_conn(state) do
+    redis_opts = Keyword.take(state.opts, @redix_opts)
+    case Redix.PubSub.start_link(redis_opts, [sync_connect: true]) do
```
Comment (chrismccord) [inline lib/phoenix_pubsub_redis/redis_server.ex]: kill the [ and ]

File: mix.exs [hunk 3]
```diff
@@ -18,12 +18,12 @@ defmodule PhoenixPubsubRedis.Mixfile do
   end
 
   def application do
-    [applications: [:logger, :redo, :poolboy]]
+    [applications: [:logger, :poolboy]]
```
Comment (chrismccord) [inline mix.exs]: redix should be listed here

Comment (mitchellhenke) [inline mix.exs]: My understanding is that redix does not run as an OTP application, but just as part of the supervision tree?

Comment (mitchellhenke) [inline mix.exs]: cool, added it

**Comment by chrismccord** [comment]
Let's point remix and redix_pubsub to official releases and we can :shipit:

**Comment by chrismccord** [comment]
We need to handle the case of https://github.com/phoenixframework/phoenix_pubsub_redis/pull/17 before release

**Comment by chrismccord** [comment]
I can take care of the PubSub.Local fixes once this is ready

**Comment by mitchellhenke** [comment]
@chrismccord looks like https://github.com/whatyouhide/redix_pubsub doesn't have a release yet.  will update afterwards

cc @whatyouhide

**Comment by whatyouhide** [comment]
Yes, me and @lexmag are literally discussing some remaining Redix stuff right now.

**Comment by mitchellhenke** [comment]
sounds good!

**Comment by mitchellhenke** [comment]
@chrismccord updated with redix and redix_pubsub

**Comment by chrismccord** [comment]
<3 <3 <3

**Comment by mitchellhenke** [comment]
glad to be able to contribute.  sorry it took as long as it did :)

**Comment by whatyouhide** [comment]
(that was much more my fault, sorry people 😄)

**Comment by mitchellhenke** [comment]
https://github.com/phoenixframework/phoenix_pubsub_redis/issues/16 and https://github.com/phoenixframework/phoenix_pubsub_redis/issues/13 should be closable now too

File: mix.exs [hunk 4]
```diff
@@ -18,12 +18,13 @@ defmodule PhoenixPubsubRedis.Mixfile do
...
   def application do
-    [applications: [:logger, :redo, :poolboy]]
+    [applications: [:logger, :poolboy, :redix, :redix_pubsub]]
   end
 
   defp deps do
-    [{:phoenix, "~> 1.1"},
-     {:redo, "~> 2.0.1"},
+    [{:phoenix_pubsub, "~> 1.0"},
+     {:redix, "~> 0.4.0"},
```
Comment (lexmag) [inline mix.exs]: We can remove this from dependencies since `redix_pubsub` depends on `redix` (also it starts application).

**Pull request merged**